modules/repository/src/test/java/RepositorySystemTestCase.java

```diff
@@ -0,0 +1,118 @@
+import org.junit.rules.TemporaryFolder;
+
+public class RepositorySystemTestCase extends AbstractMuleTestCase
+{
+
+    private static final BundleDescriptor VALIDA_BUNDLE = new BundleDescriptor.Builder()
+            .setGroupId("ant")
+            .setArtifactId("ant-antlr")
+            .setVersion("1.6")
+            .setType("jar")
+            .build();
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Rule
+    public ExpectedException expectedException = none();
+
+    @Test
+    public void existingResourceFromMaven() throws Exception
+    {
+        executeTestWithTemporaryRepo(() -> {
+            DefaultRepositoryService defaultRepositoryService = new DefaultRepositoryService();
+            File bundleFile = defaultRepositoryService.lookupBundle(VALIDA_BUNDLE);
+            assertThat(bundleFile, notNullValue());
+            assertThat(bundleFile.exists(), is(true));
+            assertThat(bundleFile.getAbsolutePath().startsWith(temporaryFolder.getRoot().getAbsolutePath()), is(true));
+        });
+    }
+
+    @Test
+    public void noExistentResource() throws Exception
+    {
+        executeTestWithTemporaryRepo(() -> {
+            DefaultRepositoryService defaultRepositoryService = new DefaultRepositoryService();
+            expectedException.expect(BundleNotFoundException.class);
+            defaultRepositoryService.lookupBundle(new BundleDescriptor.Builder()
+                                                          .setGroupId("no")
+                                                          .setArtifactId("existent")
+                                                          .setVersion("bundle")
+                                                          .setType("jar")
+                                                          .build());
+        });
+    }
+
+    @Test
+    public void invalidExternalRepository() throws Exception
+    {
+        executeTestWithTemporaryRepo(() -> {
+            testWithSystemProperty(MULE_REMOTE_REPOSITORIES_PROPERTY, "http://doesnotexists/repo", () -> {
+                DefaultRepositoryService defaultRepositoryService = new DefaultRepositoryService();
+                expectedException.expect(RepositoryConnectionException.class);
+                defaultRepositoryService.lookupBundle(VALIDA_BUNDLE);
+            });
+        });
+    }
+
+    @Test
+    public void noRepositoryConfigured() throws Exception
+    {
+        testWithSystemProperty(MULE_REMOTE_REPOSITORIES_PROPERTY, "", () -> {
+            executeTestWithTemporaryRepo(() -> {
+                DefaultRepositoryService defaultRepositoryService = new DefaultRepositoryService();
+                expectedException.expect(RepositoryServiceDisabledException.class);
+                defaultRepositoryService.lookupBundle(VALIDA_BUNDLE);
+            });
+        });
+
+    }
+
+    private void executeTestWithTemporaryRepo(TestTask test) throws Exception
```

Comment:
It would more clear if there are two different methods, one using the default MULE_REMOTE_REPOSITORY_PROPERTY and another using a custom one.